src/connection/connection.js

```diff
@@ -198,37 +192,96 @@ class Connection {
 
   /**
    * Remove stream registry after it is closed.
-   * @param {string} id identifier of the stream
+   *
+   * @param {string} id - identifier of the stream
    */
   removeStream (id) {
     this.registry.delete(id)
   }
 
   /**
    * Close the connection.
-   * @return {Promise<void>}
+   *
+   * @returns {Promise<void>}
    */
   async close () {
-    if (this.stat.status === Status.CLOSED) {
+    if (this.stat.status === CLOSED) {
       return
     }
 
     if (this._closing) {
       return this._closing
     }
 
-    this.stat.status = Status.CLOSING
+    this.stat.status = CLOSING
 
     // Close raw connection
     this._closing = await this._close()
 
     this._stat.timeline.close = Date.now()
-    this.stat.status = Status.CLOSED
+    this.stat.status = CLOSED
   }
 }
 
 /**
- * @module
- * @type {typeof Connection}
+ * @typedef {Object} Stat
+ * @property {string} direction - connection establishment direction ("inbound" or "outbound").
+ * @property {Timeline} timeline - connection relevant events timestamp.
+ * @property {string} [multiplexer] - connection multiplexing identifier.
+ * @property {string} [encryption] - connection encryption method identifier.
+ *
+ * @typedef {Object} Timeline
+ * @property {number} open - connection opening timestamp.
+ * @property {number} [upgraded] - connection upgraded timestamp.
+ * @property {number} [close]
+ *
+ * @typedef {import('./status').Status} Status
```

Comment:
can we put typedef all in the same place either top or bottom ?

Reply:
yes, moving them

Reply:
I would also just move those defs into .ts file